### ISSUE TITLE
 Current CUDA Device does not support bfloat16. Please switch dtype to float16.

### ISSUE THREAD

**Issue by benam2** [mosaicml/llm-foundry#242]
Hi,

I keep receiving this error while trying to fine tune the MPT-7b model.
The current version of `cuda is 11.0`. Is there any workaround this so i can temporarily get the model run and later add a new version of Cuda?

It raises this error while in this line in `train.py` script -> `trainer.eval()`
Thanks!

**Comment by hanlint** [comment]
Hi @benam2 , could you share the logs of what error you ran into?

**Comment by benam2** [comment]
@hanlint Sure, here is the log:

`Found cached dataset parquet (/.cache/huggingface/datasets/tatsu-lab___parquet/tatsu-lab--alpaca-2b32f0433506ef5f/0.0.0/2a3b91fbd88a2c90d1dbbb32b460cf621d31bd5b05b934492fdef7d8d6f236ec)
Loading cached processed dataset at /.cache/huggingface/datasets/tatsu-lab___parquet/tatsu-lab--alpaca-2b32f0433506ef5f/0.0.0/2a3b91fbd88a2c90d1dbbb32b460cf621d31bd5b05b934492fdef7d8d6f236ec/cache-0a7995bd07b1b853.arrow
Found cached dataset parquet (/.cache/huggingface/datasets/mosaicml___parquet/mosaicml--dolly_hhrlhf-9d0c74ee24e2b1d0/0.0.0/2a3b91fbd88a2c90d1dbbb32b460cf621d31bd5b05b934492fdef7d8d6f236ec)
Loading cached processed dataset at /.cache/huggingface/datasets/mosaicml___parquet/mosaicml--dolly_hhrlhf-9d0c74ee24e2b1d0/0.0.0/2a3b91fbd88a2c90d1dbbb32b460cf621d31bd5b05b934492fdef7d8d6f236ec/cache-8aa53017aa62d9ec.arrow
/.conda/envs/llm/lib/python3.9/site-packages/composer/callbacks/speed_monitor.py:120: UserWarning: gpu_flop count not found for None with precision: amp_bf16; MFU cannot be calculated and reported. gpu_flops_available can be manuallyoverridden by setting gpu_flops_available in SpeedMonitor.
  warnings.warn(
Traceback (most recent call last):
  File "/Desktop/llm-foundry/scripts/train/train.py", line 255, in <module>
    main(cfg)
  File "/Desktop/llm-foundry/scripts/train/train.py", line 241, in main
    trainer.eval()
  File "/.conda/envs/llm/lib/python3.9/site-packages/composer/trainer/trainer.py", line 2623, in eval
    self._eval_loop(
  File "/.conda/envs/llm/lib/python3.9/site-packages/composer/trainer/trainer.py", line 2704, in _eval_loop
    with _get_precision_context(self.state.precision, self.deepspeed_enabled):
  File "/.conda/envs/llm/lib/python3.9/contextlib.py", line 117, in __enter__
    return next(self.gen)
  File "/.conda/envs/llm/lib/python3.9/site-packages/composer/core/precision.py", line 61, in get_precision_context
    with torch.cuda.amp.autocast(enabled=True, dtype=torch.bfloat16):
  File "/.conda/envs/llm/lib/python3.9/site-packages/torch/cuda/amp/autocast_mode.py", line 25, in __init__
    super().__init__("cuda", enabled=enabled, dtype=dtype, cache_enabled=cache_enabled)
  File "/.conda/envs/llm/lib/python3.9/site-packages/torch/amp/autocast_mode.py", line 225, in __init__
    raise RuntimeError('Current CUDA Device does not support bfloat16. Please switch dtype to float16.')
RuntimeError: Current CUDA Device does not support bfloat16. Please switch dtype to float16.`

**Comment by abhi-mosaic** [comment]
Hi @benam2 , what GPU are you using? Is it a pre-Ampere card (like V100, T4)? If so, those devices do not support `bfloat16`. You will have to try finetuning with `precision: fp32` or `precision: amp_fp16` rather than the default `precision:amp_bf16`.

**Comment by benam2** [comment]
@abhi-mosaic I ended up using this code for the fine tuning and https://colab.research.google.com/drive/1DqKNPOzyMUXmJiJFvJITOahVDxCrA-wA#scrollTo=wShqQoppuv-h

And I still get the same error as I shared before. I use the G5 Ec2 instance machine (A10).
In the code above when I change `torch.bfloat16` to `32` I get the OOM error. 

Is there anyway I can fix this using the same machine? Thanks for your time
